contrib/plugins/funds/funds.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+""" This plugin gives you a nicer overview of the funds that you own.
+
+Instead of calling listfunds and adding all outputs and channels
+this plugin does that for you.
+
+Activate the plugin with: 
+`lightningd --plugin=PATH/TO/LIGHTNING/contrib/plugins/funds/funds.py`
+
+Call the plugin with: 
+`lightning-cli funds`
+
+
+Author: Rene Pickhardt (https://ln.rene-pickhardt.de)
+Development of the plugin was sponsored by fulmo.org
+"""
+
+import json
+
+from lightning.lightning import LightningRpc
+from lightning.plugin import Plugin
+from os.path import join
+
+rpc_interface = None
+plugin = Plugin(autopatch=True)
+
+
+@plugin.method("funds")
+def funds(plugin, unit="s"):
+    """Lists the total funds the lightning node owns off- and onchain in {unit}.
+
+{unit} can take the following values:
+s, satoshi, satoshis to depict satoshis
+b, bit, bits to depict bits
+m, milli, btc to depict milliBitcoin
+B, bitcoin, btc to depict Bitcoins
+
+When not using Satoshis (default) the comma values are rounded off."""
+
+    switcher = {
+        "bitcoin": "BTC",
+        "btc": "BTC",
+        "satoshi": "sat",
+        "satoshis": "sat",
+        "bit": "bit",
+        "bits": "bit",
+        "milli": "mBTC",
+        "mbtc": "mBTC",
+        "millibtc": "mBTC",
+        "B": "BTC",
+        "s": "sat",
+        "m": "mBTC",
+        "b": "bit",
+    }
+
+    if unit != "B":
+        unit = switcher.get(unit.lower(), "sat")
+    else:
+        unit = "BTC"
+
+    switcher = {
+        "sat": 1,
+        "bit": 100,
+        "mBTC": 100*1000,
+        "BTC": 100*1000*1000,
+    }
+
+    div = switcher.get(unit, 1)
+
+    funds = rpc_interface.listfunds()
+
+    onchain_value = sum([int(x["value"]) for x in funds["outputs"]])
+    offchain_value = sum([int(x["channel_sat"]) for x in funds["channels"]])
+
+    total_funds = onchain_value + offchain_value
+
+    return {
+        'total_'+unit: total_funds//div,
+        'onchain_'+unit: onchain_value//div,
+        'offchain_'+unit: offchain_value//div,
+    }
+
+
+@plugin.method("init")
+def init(options, configuration, plugin):
+    global rpc_interface
+    plugin.log("start initialization of the funds plugin")
+    print(configuration)
+    basedir = configuration['lightning-dir']
+    rpc_filename = configuration['rpc-file']
+    path = join(basedir, rpc_filename)
+    plugin.log("rpc interface located at {}".format(path))
+    rpc_interface = LightningRpc(path)
+
+    plugin.log("Funds Plugin successfully initialezed")
+
+
+plugin.add_option('funds', 'funds',
```

Comment:
This option doesn't really seem to do anything. From the description I take it that it's supposed to be the `@plugin.method` call above.

Reply:
What you could do instead is provide the default unit here, so that on startup we can chose whether we'd like to display in satoshis, bits, mBTC or BTC by default.

Another option that would make sense is the number of decimal places to show.

Reply:
Maybe I did not understand the `add_option()` call correctly. I had the feeling this was similar to the entry in the `get_manifest` that I had to be used previously. I guess I got that wrong. Will take a closer look at the implementation tomorrow

Reply:
`add_option` will add command line options to `lightningd` when starting the daemon and the option values will be passed to the plugin on `init`. So this declaration would result in a command line option `funds` with default value `funds` for the `lightningd`. So this would be perfectly fine:

```
lightningd --funds=bla
```

But then you don't extract and store the value of that option, so it's not being used at all :wink:

Reply:
Damn it I already knew that. But with the frequent api changes I oversaw that. Well certainly fix that (:

Reply:
also I am just happy that I made this mistake. I will try to change the code so that I will add an argument -`--fundsunit=s` to lightningd which would set the standard value for the command line tool. this is also good for educational purpose to see the difference and both methods